client/my-sites/plugins/controller.js

```diff
@@ -94,6 +101,30 @@ function plugin( context, next ) {
 	next();
 }
 
+export function redirectOnSearchQuery( context, next ) {
+	const searchTerm = context.query.s;
+	const site = context.params.site;
+
+	if ( searchTerm ) {
+		const redirectionUrl = `/plugins/${ site || '' }?s=${ searchTerm }`;
+		page.redirect( redirectionUrl );
+	}
+
+	next();
+}
+
+export function redirectOnUnknownCategory( context, next ) {
+	const category = context.params.category;
+	const site = context.params.site;
+
+	if ( ! ALLOWED_CATEGORIES.includes( category ) ) {
+		const redirectionUrl = `/plugins/${ site || '' }`;
+		page.redirect( redirectionUrl );
+	}
+
+	next();
+}
```

Comment:
This is meant to be a refactor PR, can we move this behavior change to a new PR?

Reply:
I'll do it

Reply:
New PR here: #66941